entity-service-impl/src/main/java/org/hypertrace/entity/query/service/converter/ValueHelper.java

```diff
@@ -157,10 +160,13 @@ public ConstantExpression convertToConstantExpression(final Value value)
 
   public SubDocumentValue convertToSubDocumentValue(final Value value) throws ConversionException {
     final ValueType type = value.getValueType();
-
     if (isArray(type)) {
       final List<Document> documents = getDocumentListFromArrayValue(value);
+      return SubDocumentValue.of(documents);
+    }
 
+    if (isMap(type)) {
+      final List<Document> documents = getDocumentsFromMapValue(value);
```

Comment:
This would just be a single Document so that it can be directly converted to JSON. Otherwise, it could result in a JSON array.